ui/signals.py

```diff
@@ -46,18 +48,37 @@ def delete_youtube_video(sender, **kwargs):
         remove_youtube_video.delay(kwargs['instance'].id)
 
 
-@receiver(post_save, sender=Video)
-def sync_youtube(sender, **kwargs):
+@receiver(post_save, sender=Collection)
+def update_collection_youtube(sender, **kwargs):
     """
+    If a collection's stream source is changed, sync YoutubeVideo objects for public Videos
+    """
+    if settings.ENABLE_VIDEO_PERMISSIONS:
+        for video in kwargs['instance'].videos.filter(is_public=True):
+            sync_youtube(video)
 
-    Upload a video to youtube if it is public and not already on YouTube.
-    Delete from youtube if it is there and permissions are not public.
+
+@receiver(post_save, sender=Video)
+def update_video_youtube(sender, **kwargs):
+    """
+    If a video's is_public field is changed, sync associated YoutubeVideo object
     """
     if settings.ENABLE_VIDEO_PERMISSIONS:
-        video = kwargs['instance']
-        yt_video_id = video.youtube_id
-        if video.is_public:
-            if not yt_video_id:
+        sync_youtube(kwargs['instance'])
+
+
+def sync_youtube(video):
+    """
+    Upload a video to youtube if it is public, not already on YouTube, and collection stream_source != cloudfront.
```

Comment:
Can you add the argument to the docstring?

Reply:
✅